### PR TITLE
Tx track modif 7087 v14

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/7087

Describe changes:
- app-layer: track modified/processed txs

#11912 with ftp-data using `updated` and trying to be smarter with some protocols only updating the right direction

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 23069

File: rust/src/applayer.rs [hunk 1]
```diff
@@ -155,6 +159,7 @@ impl AppLayerTxData {
             file_flags: 0,
             file_tx: 0,
             stream_logged: 0,
+            updated: [true; 2],
```
Comment (victorjulien) [inline rust/src/applayer.rs:162]: creating a new tx should not lead to it being updated in both dirs? Just in the dir it was created in?

File: rust/src/applayertemplate/template.rs [hunk 2]
```diff
@@ -200,6 +200,7 @@ impl TemplateState {
                     start = rem;
 
                     if let Some(tx) =  self.find_request() {
+                        tx.tx_data.updated[1] = true;
```
Comment (victorjulien) [inline rust/src/applayertemplate/template.rs:203]: I don't really like this indexing with just a number to indicate a direction. Can we either index by some name or use something like `updated_ts` / `updates_tc`

File: rust/src/applayer.rs (hunk 1 above)
Comment (catenacyber) [reply rust/src/applayer.rs:162]: right, will optimize

**Comment by catenacyber** [comment]
Next in https://github.com/OISF/suricata/pull/11966

**Pull request closed without merging**